File: c++/data_src_dependent/grid_from_bridges/grid_from_bridges.cpp
#include <string>
#include "Bridges.h"
#include "DataSource.h"
#include "ColorGrid.h"


using namespace std;
using namespace bridges;

int main(int argc, char **argv) {
#if TESTING
	// command line args provide credentials and server to test on
	Bridges bridges(atoi(argv[1]), argv[2], argv[3]);
	if (argc > 4)
		bridges.setServer(argv[4]);
#else
	Bridges bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID",
		"YOUR_API_KEY");
#endif

	//First make a ColorGrid
	int sizeI = 25, sizeJ = 35;
	ColorGrid cg1 (sizeI, sizeJ);
	for (int i = 0; i < sizeI; ++i)
		for (int j = 0; j < sizeJ; ++j) {
			if ((i + j) % 2 == 0) {
				cg1.set(i, j, Color(0, 0, 0));
			}
			else {
				cg1.set(i, j, Color(255, 255, 255));
			}
		}
	bridges.setDataStructure(&cg1);
	bridges.visualize();

	try {
		DataSource ds(&bridges);
		ColorGrid cg = ds.getColorGridFromAssignment(bridges.getUserName(), bridges.getAssignment());

		// just to be sure this is working, lets change the mid square
		// of pixels to a red colored square

		int h = cg.getHeight();
		int w = cg.getWidth();
		cout << "Height, Width:" << h << "," << w << endl;
		for (int k = h/2 - 5; k < h/2 + 5; k++)
				for (int j = w/2 - 5; j < w/2 + 5; j++)
					cg.set(k, j, Color("red"));
				bridges.setDataStructure(&cg);
				bridges.visualize();

		}
catch (char const * s) {
		std::cerr << "exception caught: " << s << std::endl;
		return -1;
	}
	catch (rapidjson_exception re) {
		std::cerr << "rapidjson_exception: " << (std::string)re << std::endl;
	}

	return 0;
}

File: java/data_src_dependent/grid_from_bridges/grid_from_bridges.java
import bridges.base.Color;
import bridges.base.ColorGrid;
import bridges.connect.Bridges;

import java.util.ArrayList;
import java.util.Arrays;

public class grid_from_bridges {


    public static void main(String[] args) throws Exception {
        Bridges bridges = new Bridges (Integer.parseInt(args[0]), args[1], args[2]);
        bridges.setServer (args[3]);


	int sizeI=25, sizeJ=35;
	ColorGrid cg1 = new ColorGrid(sizeI, sizeJ);
	
	for (int i = 0; i < sizeI; ++i)
	    for (int j = 0; j < sizeJ; ++j) {
		if ((i + j) % 2 == 0) {
		    cg1.set(i, j, new Color(0, 0, 0));
		}
		else {
		    cg1.set(i, j, new Color(255, 255, 255));
		}
	    }
	bridges.setDataStructure(cg1);
	bridges.visualize();

	ColorGrid cg= bridges.getColorGridFromAssignment(bridges.getUserName(), bridges.getAssignmentID(), 0);

	int[] dims = cg.getDimensions();
	for (int k = dims[1] / 2 - 5; k < dims[1] / 2 + 5; k++)
	    for (int j = dims[0] / 2 - 5; j < dims[0] / 2 + 5; j++)
		cg.set(k, j, new Color("red"));

	bridges.setDataStructure(cg);
	bridges.visualize();
	
	
    }
}
